servicex/servicex_utils.py

```diff
@@ -21,12 +21,15 @@ def _wrap_in_memory_sx_cache(fn):
     '''
     @functools.wraps(fn)
     async def cached_version_of_fn(*args, **kwargs):
-        assert len(args) == 2
+        assert len(args) == 2 or 3
```

Comment:
Python does not interpret `len(args) == 2 or 3` the way you might think. It will read it as `(len(args) == 2) or (3)`. When there are 2 arguments, this expression evaluates to `True`, but when the number of arguments is not 2, the expression evaluates to `3` (the `or` operator simply returns the first truthy expression). This is not causing an error because `assert 3` is valid Python.

Should be changed to one of the options below to get the desired effect:
- `assert len(args) in (2, 3)`
- `assert 2 <= len(args) <= 3`
- `assert len(args) == 2 or len(args) == 3`

Reply:
@AndrewEckart - you are absolutely right! I will fix the bug.